plugnpy/metric.py

```diff
@@ -84,24 +89,48 @@ def __init__(
         self.convert_metric = convert_metric
 
         self.si_bytes_conversion = si_bytes_conversion
-        bytes_conversion_factor = Metric.SI_UNIT_FACTOR if si_bytes_conversion else Metric.IEC_UNIT_FACTOR
-        self.conversion_factor = bytes_conversion_factor if self.unit == Metric.UNIT_BYTES else Metric.SI_UNIT_FACTOR
+        if self.unit == Metric.UNIT_BYTES and not si_bytes_conversion:
+            self.conversion_factor = Metric.IEC_UNIT_FACTOR
+        else:
+            self.conversion_factor = Metric.SI_UNIT_FACTOR
 
-        self.precision = precision
+        try:
+            self.precision = int(precision)
+        except ValueError as ex:
+            raise Exception("Invalid value for precision '{0}': {1}".format(self.precision, ex))
 
         self.warning_threshold = warning_threshold
         self.critical_threshold = critical_threshold
+        self.state = Metric.evaluate(value, warning_threshold, critical_threshold, self.conversion_factor)
 
-        if self.critical_threshold and self.check(value, *self.parse_threshold(self.critical_threshold)):
-            self.state = 2
-        elif self.warning_threshold and self.check(value, *self.parse_threshold(self.warning_threshold)):
-            self.state = 1
-        else:
-            self.state = 0
+    def __str__(self):
+        if self.message:
+            return self.message
+        name = self.display_name
+        unit = self.unit
+        value = self.value
+
+        if self.convert_metric:
+            value, unit = Metric.convert_value(self.value, self.unit, self.conversion_factor)
+
+        value = "{0:.{1}f}".format(value, self.precision)
+
+        return self.display_format.format(**{'name': name, 'value': value, 'unit': unit})
 
+    @staticmethod
+    def evaluate(value, warning_threshold, critical_threshold, conversion_factor):
```

Comment:
conversion factor should not be required + explain this method in the docs? Move to **si_bytes_conversion** and make it true/false